var/spack/repos/builtin/packages/cryptsetup/package.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Cryptsetup(AutotoolsPackage):
+    """Cryptsetup and LUKS - open-source disk encryption."""
+
+    homepage = "https://gitlab.com/cryptsetup/cryptsetup"
+    url      = "https://www.kernel.org/pub/linux/utils/cryptsetup/v2.2/cryptsetup-2.2.1.tar.xz"
+
+    version('2.2.1', sha256='94e79a31ed38bdb0acd9af7ccca1605a2ac62ca850ed640202876b1ee11c1c61')
+
+    depends_on('libuuid', type=('build', 'link'))
+    depends_on('lvm2', type=('build', 'link'))
+    depends_on('popt', type=('build', 'link'))
+    depends_on('json-c', type=('build', 'link'))
+    depends_on('util-linux', type=('build', 'link'))
+    depends_on('gettext', type=('build', 'link'))
+
+    def configure_args(self):
+        args = [
+            'LIBS=-lintl',
```

Comment:
What happens if you instead add

```
--with-libintl-prefix[=DIR]
```

to `configure_args`, with `DIR` being something derived from `self.spec['gettext'].prefix.sbin` (-ish?)?

Reply:
Are you asking just out of curiosity?  Is this somehow superior to `LIBS=-lintl`?  Because this already seems to solve the issue I was having earlier.

Reply:
Sorry for being vague.  The `LIBS=-lintl` seems like a hack to work around something that's not working as it should.  Might be that the configure script is broken, but it also might be that the right levers aren't being pulled.

Reply:
Here's an alternate fix for the problem, that's perhaps better (I prefer it, but...):

https://github.com/hartzell/spack/commit/02ecddf028d3d4aec7d330fd2990e0f2efd860e7

The problem is that cryptsetup does all of the Proper Autotools Things(tm) to discover that it needs to add `-lintl` to the LDFLAGS, but doesn't actually make use of the information.   The patch in the commit above adds the value of LTLIBINTL (which configure populates for us) to AM_LDFLAGS, which eventually find its way into the LDFLAGS, et viola!  It requires that the `autoreconf` step run, so it also adds the necessary dependencies.

I'm confident of the diagnosis.  I'm fairly comfortable that using AM_LDFLAGS is the right way to do it, but someone with add'l autotools knowledge might know a better way.

Perhaps someone should upstream this, but I'm running low on volunteer TUITS....

Reply:
@alalazo If you want to 👍 the above solution, I will apply it to this PR.
@hartzell Thanks much!  I see now why you didn't like the smell of the configure arg.  Does the verb "upstream"  in this context mean: open a PR with cryptsetup?  I would be willing to do that, if it's okay with you.

Reply:
@odoublewen -- yes, upstream would be a bug in the cryptsetup project.  Please help yourself!  If/when you do, you should update the comment above the patch to reference the bug, if/when it's fixed we'll need to constrain the `patch()` statement (expression?).  It'd also be interesting to keep an eye on it and see if there's anyone "upstream" who has better Automake-fu and has a different solution.

I'm comfortable enough with this for now though.

Reply:
pps @odoublewen -- you might also notice that this change requires us to make the pizza from scratch, but it appears to work simply, so....

Reply:
OK, I've merged your edit into my fork, and then I added a comment line referencing the issue that I opened with cryptsetup.